### PR TITLE
Support log configuration for cloudwatch.

### PR DESCRIPTION
**Pull request by pilhuhn**
Also if certain root elements are missing, but requested, throw an IllegalStateException.

### CODE REVIEW

File: src/test/java/com/redhat/cloud/common/clowder/configsource/ConfigSourceTest2.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ *
+ */
+public class ConfigSourceTest2 {
```
Comment (bennetelli) [inline src/test/java/com/redhat/cloud/common/clowder/configsource/ConfigSourceTest2.java:18]: I would like to rename it to a more (domain) specific name or even put it as @Nested class in the same class as ConfigSourceTest. This test class looks a bit copy-pasted ;-)

Comment (bennetelli) [reply src/test/java/com/redhat/cloud/common/clowder/configsource/ConfigSourceTest2.java:18]: .

**Review by bennetelli — CHANGES_REQUESTED** [review]

File: src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java [hunk 2]
```diff
@@ -114,10 +119,13 @@ public String getValue(String configKey) {
             }
 
             if (configKey.startsWith("mp.messaging") && configKey.endsWith(".topic")) {
+                if (kafkaBase==null) {
```
[the remark this answers is not in the document]
Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java:122]: You mean after line 103? That would require a Kafka section even in projects that don't use Kafka.

File: src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java [hunk 3]
```diff
@@ -99,8 +100,12 @@ public String getValue(String configKey) {
                 JsonNumber webPort = root.getJsonNumber("webPort");
                 return webPort.toString();
             }
+            JsonObject kafkaBase = root.getJsonObject("kafka");
             if (configKey.equals("kafka.bootstrap.servers")) {
-                JsonArray brokers = root.getJsonObject("kafka").getJsonArray("brokers");
+                if (kafkaBase==null) {
```
Comment (bennetelli) [inline src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java:107]: can't you just put line 105-107 between 103 and 104?

**Review by bennetelli — APPROVED** [review]

**Review by bennetelli — APPROVED** [review]

**Pull request merged**